gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/api/AbstractConfigurablePluginManager.java

```diff
@@ -74,21 +91,33 @@ private String getFileFromPropertyAsBase64(T plugin, Map<String, String> propert
 
     @Override
     public String getDocumentation(String pluginId) throws IOException {
-        return getFirstFile(pluginId, DOCS_DIRECTORY);
+        return getFirstFile(pluginId, DOCS_DIRECTORY, false);
+    }
+
+    @Override
+    public String getDocumentation(String pluginId, boolean includeNotDeployed) throws IOException {
+        return getFirstFile(pluginId, DOCS_DIRECTORY, includeNotDeployed);
     }
 
     @Override
     public String getCategory(String pluginId) throws IOException {
-        T plugin = get(pluginId);
-        Map<String, String> properties = plugin.manifest().properties();
-        if (properties != null) {
-            return properties.get(PluginManifestProperties.MANIFEST_CATEGORY_PROPERTY);
+        return getCategory(pluginId, false);
+    }
+
+    @Override
+    public String getCategory(String pluginId, boolean includeNotDeployed) throws IOException {
+        T plugin = get(pluginId, includeNotDeployed);
+        if (plugin != null) {
+            Map<String, String> properties = plugin.manifest().properties();
+            if (properties != null) {
+                return properties.get(PluginManifestProperties.MANIFEST_CATEGORY_PROPERTY);
+            }
         }
         return null;
```

Comment:
I see a lot of null checks. Can we work with `Optional` ?
Does the `get (pluginId, includeNotDeployed)` is called somewhere else not to use optionals (if we can use `Optional.ofNullable` to wrap it around) ?

We could have this method returning (considering `get` returns an Optional)

```suggestion
        return get(pluginId, includeNotDeployed)
                .map(Plugin::manifest)
                .map(PluginManifest::properties)
                .map(properties -> properties.get(PluginManifestProperties.MANIFEST_CATEGORY_PROPERTY))
                .orElse(null);
```
That way we improve readability and ensure null checking

Reply:
Yes unfortunately, the `get` method is called a lot so we cannot replace it easily. I looked at the other methods that could benefit from this change but there are special things like unchecked exception thrown that prevent from doing it so I prefer for the moment to keep the consistency in the class and let it this way.